### PR TITLE
[docs] Update relay docs

### PR DESCRIPTION
**Pull request by zhiics**
#5068 

This PR updates Relay docs. It removes redundant files and automatically generates automodules. 

cc @tqchen @comaniac @cchung100m

### CODE REVIEW

File: docs/api/python/relay/index.rst [hunk 1]
```diff
@@ -27,16 +27,22 @@ compiler stack.
...
    nn
    op
-   scope_builder
    vision
    testing
+
+.. automodule:: tvm.relay
+    :members:
+    :imported-members:
+    :exclude-members: RelayExpr, Pass, PassInfo, function_pass, PassContext,
```
Comment (tqchen) [inline docs/api/python/relay/index.rst]: We can also consider allow these members, but modify https://github.com/apache/incubator-tvm/blob/master/docs/conf.py#L254 to mark them as alias

File: tutorials/relay_quick_start.py [hunk 2]
```diff
@@ -79,14 +79,14 @@
 # Currently this value can be 0 to 3. The optimization passes include
 # operator fusion, pre-computation, layout transformation and so on.
 #
-# :any:`relay.build_module.build` returns three components: the execution graph in
+# :any:`relay.build` returns three components: the execution graph in
```
Comment (tqchen) [inline tutorials/relay_quick_start.py]: `:any:->:py:func:`

File: docs/api/python/relay/index.rst (hunk 1 above)
Comment (zhiics) [reply docs/api/python/relay/index.rst]: Yeah, I tried this before. But there are a lot of places that directly use the classes without namespace. It causes many ambiguous reference. For example:

https://github.com/apache/incubator-tvm/blob/master/python/tvm/ir/adt.py#L82

**Comment by zhiics** [comment]
@tqchen BTW, the CI still complains about the conflict from `relay_op.rst`. I couldn't repro it locally. Is it due to some caching problem?

**Comment by tqchen** [comment]
interesting, can you try to use the exact same docker image? Iet us also try to add make clean before https://github.com/apache/incubator-tvm/blob/master/tests/scripts/task_sphinx_precheck.sh#L36

**Comment by zhiics** [comment]
I checked out the image and ran the same command before. It seemed working fine. Hopefully I didn't do something silly. I will double check.

**Comment by zhiics** [comment]
@tqchen Yeah, it should be cached before. `make clean` solve the problem. Thanks.

**Comment by tqchen** [comment]
Great, let us also add make clean to https://github.com/apache/incubator-tvm/blob/master/tests/scripts/task_python_docs.sh#L47

**Pull request merged**